sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/EliminateResolvedHint.scala

```diff
@@ -29,7 +29,7 @@ object EliminateResolvedHint extends Rule[LogicalPlan] {
   // is using transformUp rather than resolveOperators.
   def apply(plan: LogicalPlan): LogicalPlan = {
     val pulledUp = plan transformUp {
-      case j: Join =>
+      case j: Join if j.hint == JoinHint.NONE =>
```

Comment:
@maryannxue If we don't do this transformation, should we also skip the next one that removes the `ResolvedHint` or you think thats not required ?

Reply:
I think the "best" way to fix it is to merge the new hints into the existing `JoinHint`, but the question is which one should take precedence, the new one or the existing one?
So far, if things all work right, we should never come to a point where there is a non-empty new hint and a non-empty existing hint, that is, the hint has either been pulled into the join node or it hasn't been processed it. In either case, this fix is sufficient.
The removal of the `ResolvedHint` is taken care of by the following lines.

Reply:
@maryannxue Thanks for explanation.